upload/src/addons/SV/WarningImprovements/XF/Entity/Warning.php

```diff
@@ -35,6 +36,43 @@ public function canViewIssuer()
         return $visitor->canViewIssuer();
     }
 
+    public function getAnonymizedIssuer()
+    {
+        $anonymizedIssuer = $this->em()->create('XF:User');
+        $anonymizedIssuer->username = \XF::phrase('WarningStaff')->render();
```

Comment:
Use `$this->repository('XF:User')->getGuestUser()` instead of manually creating the user object.

This should be done after the Warning Staff can't be found if $anonymizedIssuer is null.

Reply:
Done